src/borg/archive.py

```diff
@@ -1142,7 +1164,7 @@ def item_is_hardlink_master(item):
                     'source' not in item and
                     not matcher.match(item.path))
 
-        for item in archive.iter_items():
+        for item in archive.iter_items(filter=lambda item: 'checkpoint' not in item):
```

Comment:
It would make sense to move this check to iter_items() since this conditions is always used.